File: CS/WinWebSolution.Module/GaugeDemoObject.cs
using System;
using DevExpress.Xpo;
using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace WinWebSolution.Module {
    [DefaultClassOptions]
    public class GaugeDemoObject : BaseObject {
        public GaugeDemoObject(Session session) : base(session) { }
        private string _stringProperty;
        public string StringProperty {
            get { return _stringProperty; }
            set { SetPropertyValue("StringProperty", ref _stringProperty, value); }
        }
        private int _intProperty;
        public int IntProperty {
            get { return _intProperty; }
            set { SetPropertyValue("IntProperty", ref _intProperty, value); }
        }
        private float _floatProperty;
        public float FloatProperty {
            get { return _floatProperty; }
            set { SetPropertyValue("FloatProperty", ref _floatProperty, value); }
        }
        private Light _enumProperty;
        public Light EnumProperty {
            get { return _enumProperty; }
            set { SetPropertyValue("EnumProperty", ref _enumProperty, value); }
        }
    }
    public enum Light {
        Red,
        Yellow,
        Green
    }
}

File: VB/WinWebSolution.Module/GaugeDemoObject.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace WinWebSolution.Module
	<DefaultClassOptions> _
	Public Class GaugeDemoObject
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _stringProperty As String
		Public Property StringProperty() As String
			Get
				Return _stringProperty
			End Get
			Set(ByVal value As String)
				SetPropertyValue("StringProperty", _stringProperty, value)
			End Set
		End Property
		Private _intProperty As Integer
		Public Property IntProperty() As Integer
			Get
				Return _intProperty
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("IntProperty", _intProperty, value)
			End Set
		End Property
		Private _floatProperty As Single
		Public Property FloatProperty() As Single
			Get
				Return _floatProperty
			End Get
			Set(ByVal value As Single)
				SetPropertyValue("FloatProperty", _floatProperty, value)
			End Set
		End Property
		Private _enumProperty As Light
		Public Property EnumProperty() As Light
			Get
				Return _enumProperty
			End Get
			Set(ByVal value As Light)
				SetPropertyValue("EnumProperty", _enumProperty, value)
			End Set
		End Property
	End Class
	Public Enum Light
		Red
		Yellow
		Green
	End Enum
End Namespace